p3/metrics/_pp.py

```diff
@@ -83,6 +83,12 @@ def pp(df):
         if not df[eff].fillna(0).between(0, 1).all():
             raise ValueError("%s must in range [0, 1]" % eff)
 
+    # Keep only the most efficient (application, platform) results.
+    key = ["problem", "platform", "application"]
```

Comment:
I'm not 100% sure, but I think the `groupby` might be done in this sequence (i.e. behavior might vary with permutation). Does it matter, and if it does, can you double check?

Reply:
You're right that `groupby` behaves this way, and I didn't know that.  Thanks for pointing it out.

I don't think it matters in this case.  I just tested all the different permutations of keys, and the unit tests all pass.  I think this might be because we flatten things again a few lines later (with `reset_index(inplace=True)`), so the fact that the grouping produced a different nested index is lost.  The ordering of the keys _does_ affect the order of the columns in the output dataframe, but since we always access columns by name, I don't think that has any impact.

Reply:
Sounds good!